main.py

```diff
@@ -114,28 +115,33 @@ def run_tests(
     parse_only: bool,
     timeout: int,
     memory_limit: int,
-    on_progress_change: Callable[[int], None] | None,
+    on_progress_change: Callable[[int, dict[str, int]], None] | None,
     forward_stderr: Callable[[str], None] | None,
 ) -> list[TestRun]:
 
     current_test = 0
     results = []
 
     def add_result(
+        iteration_results: List[TestRun],
```

Comment:
Can use the built-in list here, typing.List isn't even imported :)

Reply:
thanks python, your type-checking abilities are truly amazing

Reply:
On that note, should we be running mypy or something on this?